modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1403,7 +1405,25 @@ private Publisher<BinaryRow> scan(
                         columnsToInclude,
                         implicit
                 ),
-                (commit, fut) -> postEnlist(fut, commit, actualTx, implicit && !commit)
+                (intentionallyClose, fut) -> {
+                    CompletableFuture<Void> opFut;
+
+                    if (implicit) {
+                        opFut = fut.thenApply(cursorId -> null);
```

Comment:
So, we won't call cursors.onClose if it was an implicit transaction? Why?

Reply:
Because the implicit transaction has to close the cursor in the post-enlist logic.